osu.Game.Rulesets.Catch.Tests/TestSceneHyperDash.cs

```diff
@@ -49,39 +53,50 @@ protected override IBeatmap CreateBeatmap(RulesetInfo ruleset)
             };
 
             // Should produce a hyper-dash (edge case test)
-            beatmap.HitObjects.Add(new Fruit { StartTime = 1816, X = 308 / 512f, NewCombo = true });
-            beatmap.HitObjects.Add(new JuiceStream { StartTime = 2008, X = 56 / 512f, });
+            beatmap.HitObjects.Add(new Fruit { StartTime = 1816, X = 56 / 512f, NewCombo = true });
+            beatmap.HitObjects.Add(new Fruit { StartTime = 2008, X = 308 / 512f, NewCombo = true });
 
             double startTime = 3000;
 
             const float left_x = 0.02f;
             const float right_x = 0.98f;
 
-            createObjects(() => new Fruit(), left_x);
-            createObjects(() => new JuiceStream(), right_x);
-            createObjects(() => new JuiceStream(), left_x);
-            createObjects(() => new Fruit(), right_x);
-            createObjects(() => new Fruit(), left_x);
-            createObjects(() => new Fruit(), right_x);
-            createObjects(() => new JuiceStream(), left_x);
+            createObjects(() => new Fruit { X = left_x });
+            createObjects(() => new TestJuiceStream(right_x), 1);
+            createObjects(() => new TestJuiceStream(left_x), 1);
+            createObjects(() => new Fruit { X = right_x });
+            createObjects(() => new Fruit { X = left_x });
+            createObjects(() => new Fruit { X = right_x });
+            createObjects(() => new TestJuiceStream(left_x), 1);
 
             return beatmap;
 
-            void createObjects(Func<CatchHitObject> createObject, float x)
+            void createObjects(Func<CatchHitObject> createObject, int count = 3)
             {
                 const float spacing = 140;
 
-                for (int i = 0; i < 3; i++)
+                for (int i = 0; i < count; i++)
                 {
                     var hitObject = createObject();
-                    hitObject.X = x;
                     hitObject.StartTime = startTime + i * spacing;
-
                     beatmap.HitObjects.Add(hitObject);
                 }
 
                 startTime += 700;
             }
         }
+
+        private class TestJuiceStream : JuiceStream
+        {
+            public TestJuiceStream(float x)
+            {
+                X = x;
+                Path = new SliderPath(new[]
+                {
+                    new PathControlPoint(new Vector2(x, 0)),
+                    new PathControlPoint(new Vector2(x + 30, 0)),
```

Comment:
Slightly confused as to why these use `x` at all but I suppose it doesn't *hurt*?

Reply:
It is actually wrong, it means that the path has an implicit empty linear segment from (0, 0) to (x, 0).